### ISSUE TITLE
graphql-java-servlet is incompatible with graphql-java-annotations?

### ISSUE THREAD

**Issue by mushketyk** [graphql-java/graphql-java-servlet#38]
I've registered GraphQL java servlet as defined in a workaround to:

[https://github.com/graphql-java/graphql-spring-boot/issues/20](https://github.com/graphql-java/graphql-spring-boot/issues/20)

and registered the following query root using `graphql-java-annotations`:

```java
@GraphQLName("QueryRoot")
class QueryRoot {
   @GraphQLField
    List<Item> items(DataFetchingEnvironment environment) {
       ...
    }

    @GraphQLField
    Item item(DataFetchingEnvironment environment, @GraphQLName("id") String id) {
        ...
    }
}
```

But when I perform a query I receive the following: 

```
java.lang.IllegalArgumentException: object is not an instance of declaring class
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_131]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_131]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_131]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_131]
	at graphql.annotations.MethodDataFetcher.get(MethodDataFetcher.java:63) ~[graphql-java-annotations-3.0.3.jar:na]
	at graphql.execution.ExecutionStrategy.resolveField(ExecutionStrategy.java:99) ~[graphql-java-annotations-3.0.3.jar:na]
	at graphql.execution.SimpleExecutionStrategy.execute(SimpleExecutionStrategy.java:19) [graphql-java-annotations-3.0.3.jar:na]
	at graphql.execution.Execution.executeOperation(Execution.java:106) [graphql-java-annotations-3.0.3.jar:na]
	at graphql.execution.Execution.execute(Execution.java:49) [graphql-java-annotations-3.0.3.jar:na]
	at graphql.GraphQL.execute(GraphQL.java:222) [graphql-java-3.0.0.jar:na]
	at graphql.servlet.GraphQLServlet.query(GraphQLServlet.java:277) [graphql-java-servlet-4.1.0.jar:na]
	at graphql.servlet.GraphQLServlet.lambda$new$1(GraphQLServlet.java:185) [graphql-java-servlet-4.1.0.jar:na]
	at graphql.servlet.GraphQLServlet.doRequest(GraphQLServlet.java:222) [graphql-java-servlet-4.1.0.jar:na]
	at graphql.servlet.GraphQLServlet.doPost(GraphQLServlet.java:240) [graphql-java-servlet-4.1.0.jar:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:661) ~[tomcat-embed-core-8.5.15.jar:8.5.15]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[tomcat-embed-core-8.5.15.jar:8.5.15]
```

This boils down to the following code in the `graphql-java-annotations`:

```java
class MethodDataFetcher implements DataFetcher {
  ...

    @Override
    public Object get(DataFetchingEnvironment environment) {
        try {
            Object obj;

            if (Modifier.isStatic(method.getModifiers())) {
                obj = null;
            } else if (method.getAnnotation(GraphQLInvokeDetached.class) == null) {
                obj = environment.getSource(); // <- THIS BRANCH IS EXECUTED
                if (obj == null) {
                    return null;
                }
            } else {
                obj = newInstance(method.getDeclaringClass());
            }
            //
            // FAILS HERE!
            // Expects "obj" to be of the "QueryRoot" type, but it is of the type
            // "GraphQLContext"
            return method.invoke(obj, invocationArgs(environment));
        } catch (IllegalAccessException | InvocationTargetException e) {
            throw new RuntimeException(e);
        }
    }
}
```

There are two workarounds for this:
* Annotate methods in `QueryRoot` with `@GraphQLInvokeDetached`
* Make methods in `QueryRoot` static

Is this type mismatch expected or is it a bug? It seems like it should would work without these workarounds, according to `graphql-java-annotations` [documentation](https://github.com/graphql-java/graphql-java-annotations)

**Comment by apottere** [comment]
This is a duplicate of #28.  There's an issue on graphql-java linked in that thread that has been fixed, but 4.0.0 hasn't been released yet.  Once that's released, I'll be able to fix this.

**Comment by apottere** [comment]
I'm releasing the fix for https://github.com/graphql-java/graphql-spring-boot/issues/20 today, though.
